plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go

```diff
@@ -166,11 +166,21 @@ func defaultPredicates() sets.String {
 		// Fit is determined based on whether a pod can tolerate all of the node's taints
 		factory.RegisterFitPredicate("PodToleratesNodeTaints", predicates.PodToleratesNodeTaints),
 
-		// Fit is determined by volume zone requirements.
+		// Fit is determined by volume node requirements.
+		// This predicate has been replaced by the VolumeBinding predicate but it needs to remain registered
+		// for backwards compatibility
 		factory.RegisterFitPredicateFactory(
 			"NoVolumeNodeConflict",
 			func(args factory.PluginFactoryArgs) algorithm.FitPredicate {
-				return predicates.NewVolumeNodePredicate(args.PVInfo, args.PVCInfo, nil)
+				return predicates.NewVolumeNodePredicate(args.PVInfo, args.PVCInfo)
+			},
+		),
+
+		// Fit is determined by volume topology requirements.
+		factory.RegisterFitPredicateFactory(
```

Comment:
See if this can be registered only if feature gate is enabled.  Then we also don't have to add it to the compatibility test.